src/wallet/test/group_outputs_tests.cpp

```diff
@@ -0,0 +1,224 @@
+    for (unsigned long i = 0; i < GROUP_SIZE; i++) {
+        addCoin(group_verifier.coins_pool, *wallet, dest, 10 * COIN, /*is_from_me=*/true);
+    }
+
+    group_verifier.GroupAndVerify(BASIC_FILTER,
+                                  /*expected_with_partial_spends_size=*/ GROUP_SIZE,
+                                  /*expected_without_partial_spends_size=*/ 1,
+                                  /*positive_only=*/ true);
+
+    // ####################################################################################
+    // 3) 10 more UTXO are added with a different script --> must be grouped into a single
+    //    group for avoid partial spends and 10 different output groups for partial spends
+    // ####################################################################################
+
+    const CTxDestination dest2 = *Assert(wallet->GetNewDestination(OutputType::BECH32, ""));
+    for (unsigned long i = 0; i < GROUP_SIZE; i++) {
+        addCoin(group_verifier.coins_pool, *wallet, dest2, 5 * COIN, /*is_from_me=*/true);
+    }
+
+    group_verifier.GroupAndVerify(BASIC_FILTER,
+            /*expected_with_partial_spends_size=*/ GROUP_SIZE * 2,
+            /*expected_without_partial_spends_size=*/ 2,
+            /*positive_only=*/ true);
+
+    // ################################################################################
+    // 4) Now add a negative output --> which will be skipped if "positive_only" is set
+    // ################################################################################
+
+    const CTxDestination dest3 = *Assert(wallet->GetNewDestination(OutputType::BECH32, ""));
+    addCoin(group_verifier.coins_pool, *wallet, dest3, 1, true, CFeeRate(100));
+    BOOST_CHECK(group_verifier.coins_pool.coins[OutputType::BECH32].back().GetEffectiveValue() <= 0);
+
+    // First expect no changes with "positive_only" enabled
+    group_verifier.GroupAndVerify(BASIC_FILTER,
+            /*expected_with_partial_spends_size=*/ GROUP_SIZE * 2,
+            /*expected_without_partial_spends_size=*/ 2,
+            /*positive_only=*/ true);
+
+    // Then expect changes with "positive_only" disabled
+    group_verifier.GroupAndVerify(BASIC_FILTER,
+            /*expected_with_partial_spends_size=*/ GROUP_SIZE * 2 + 1,
+            /*expected_without_partial_spends_size=*/ 3,
+            /*positive_only=*/ false);
+
+
+    // ##############################################################################
+    // 5) Try to add a non-eligible UTXO (due not fulfilling the min depth target for
+    //    "not mine" UTXOs) --> it must not be added to any group
+    // ##############################################################################
+
+    const CTxDestination dest4 = *Assert(wallet->GetNewDestination(OutputType::BECH32, ""));
+    addCoin(group_verifier.coins_pool, *wallet, dest4, 6 * COIN,
+            /*is_from_me=*/false, CFeeRate(0), /*depth=*/2);
```

Comment:
nit: could use the just-below-boundary value of 5 here
```suggestion
            /*is_from_me=*/false, CFeeRate(0), /*depth=*/5);
```

Reply:
done